cli/azd/internal/appdetect/appdetect.go

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package appdetect
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/bmatcuk/doublestar/v4"
+)
+
+type Language string
+
+const (
+	DotNet     Language = "dotnet"
+	Java       Language = "java"
+	JavaScript Language = "js"
+	TypeScript Language = "ts"
+	Python     Language = "python"
+)
+
+func (pt Language) Display() string {
+	switch pt {
+	case DotNet:
+		return ".NET"
+	case Java:
+		return "Java"
+	case JavaScript:
+		return "JavaScript"
+	case TypeScript:
+		return "TypeScript"
+	case Python:
+		return "Python"
+	}
+
+	return ""
+}
+
+type Dependency string
+
+const (
+	JsReact   Dependency = "react"
+	JsAngular Dependency = "angular"
+	JsVue     Dependency = "vuejs"
+	JsJQuery  Dependency = "jquery"
+
+	PyFlask   Dependency = "flask"
+	PyDjango  Dependency = "django"
+	PyFastApi Dependency = "fastapi"
+)
+
+func (f Dependency) Language() Language {
+	switch f {
+	case JsReact, JsAngular, JsVue, JsJQuery:
+		return JavaScript
+	}
+
+	return ""
+}
+
+func (f Dependency) Display() string {
+	switch f {
+	case JsReact:
+		return "React"
+	case JsAngular:
+		return "Angular"
+	case JsVue:
+		return "Vue.js"
+	case JsJQuery:
+		return "JQuery"
+	}
+
+	return ""
+}
+
+func (f Dependency) IsWebUIFramework() bool {
+	switch f {
+	case JsReact, JsAngular, JsVue, JsJQuery:
+		return true
+	}
+
+	return false
+}
+
+// A type of database that is inferred through heuristics while scanning project information.
+type DatabaseDep string
+
+const (
```

Comment:
Do we need cosmos here?

Reply:
Will do this as a separate change if needed.